src/payment/strategies/googlepay/googlepay-payment-strategy.spec.ts

```diff
@@ -291,6 +291,32 @@ describe('GooglePayPaymentStrategy', () => {
                 },
             });
 
+            await strategy.initialize(googlePayOptions);
+            await expect(strategy.execute(getGoogleOrderRequestBody())).rejects.toThrow(MissingDataError);
+            expect(googlePayPaymentProcessor.displayWallet).toBeCalled();
+        });
+
+        it('gets again the payment information and get a new nonce', async () => {
+            jest.spyOn(googlePayPaymentProcessor, 'handleSuccess').mockReturnValue(Promise.resolve());
+            jest.spyOn(googlePayPaymentProcessor, 'displayWallet').mockReturnValue(getGooglePaymentDataMock());
+            jest.spyOn(store.getState().paymentMethods, 'getPaymentMethodOrThrow').mockReturnValueOnce({
+                initializationData: {
+                    nonce: '',
+                    card_information: {
+                        type: 'type',
+                        number: 'number',
+                    },
+                },
+            }).mockReturnValue({
+                initializationData: {
+                    nonce: 'newNonce',
+                    card_information: {
+                        type: 'type',
+                        number: 'number',
+                    },
+                },
+            });
+
```

Comment:
This checks for `displayWallet` to have been called. That will happen unconditionally when we call execute with an empty `nonce`. I'm not sure what this test is trying to prove.

Reply:
I try to check the flow when the `_getPayment` returns a empty nonce and enter into the function to display the wallet

Reply:
The expect in this test is `displayWallet` to have been called. Display wallet is called with and without nonce, the only reason this test fails when the nonce is not present is because some code _after_ display wallet blows up.

We shouldn't be relying on other parts of the code for the check, we should be validating the _nonce_ used is the one provided as `mockReturnValue`: `newNonce`

Reply:
I add a new expect when the  when we have this error the method `getPaymentMethodOrThrow` should be called 3 times, this assert just happen when the payment method fails and we need to show the wallet again, let me know if this assertion sounds more sense for you or we need to test some else @icatalina 🙏🏼